src/Workspaces/Core/Desktop/Workspace/Host/Documentation/DocumentationProviderServiceFactory.cs

```diff
@@ -29,15 +30,46 @@ public DocumentationProvider GetDocumentationProvider(string assemblyPath)
                     throw new ArgumentNullException(nameof(assemblyPath));
                 }
 
-                assemblyPath = Path.ChangeExtension(assemblyPath, "xml");
+                string xmlDocumentationFilePath;
+                if (!TryFindDocumentationFile(assemblyPath, out xmlDocumentationFilePath))
+                {
+                    return _assemblyPathToDocumentationProviderMap.GetOrAdd(nameof(DocumentationProvider.Default), DocumentationProvider.Default);
```

Comment:
The key here should be the path to the file, not the name of the provider, right?